checkov/secrets/scan_git_history.py

```diff
@@ -61,8 +74,64 @@ def scan_history(root_folder: str, secrets: SecretsCollection) -> None:
-                    f'{commit_hash}-{secret.filename}-{secret.secret_hash}-{"added" if secret.is_added else "removed"}'].add(secret)
+                set_secret_map(file_results, secret_map, file_name, commit_hash)
             scanned_file_count += 1
+    for secrets_data in secret_map.values():
+        for secret_data in secrets_data:
+            removed = secret_data["removed_commit_hash"] if secret_data["removed_commit_hash"] else GIT_HISTORY_NOT_BEEN_REMOVED
+            key = f'{secret_data["added_commit_hash"]}_{removed}_{secret_data["potential_secret"].filename}'
+            secrets[key].add(secret_data["potential_secret"])
     logging.info(f"Scanned {scanned_file_count} git history files")
+
+
+def get_added_and_removed_commit_hash(
+        key: str, enable_git_history_secret_scan: bool, secret: PotentialSecret) -> Tuple[str | None, str | None]:
+    """
+    now we have only the current commit_hash - in the added_commit_hash or in the removed_commit_hash.
+    in the next step we will add the connection and the missing data
+    The key is built like this:
+    '{added_commit_hash}_{removed_commit_hash or the string SECRET_NOT_BEEN_REMOVED if the secret not been removed}_{file_name}'
+    """
+    if not enable_git_history_secret_scan:
+        return None, None
+    added_commit_hash = key.split('_')[0]
+    removed_commit_hash = key.split('_')[1] if key.split('_')[1] != GIT_HISTORY_NOT_BEEN_REMOVED else None
+    return added_commit_hash, removed_commit_hash
+
+
+def set_secret_map(file_results: List[PotentialSecret], secret_map: Dict[str, List[EnrichedPotentialSecret]],
+                   file_name: str, commit_hash: str) -> None:
+    # First find if secret was moved in the file
+    equal_secret_in_commit: Dict[str, List[str]] = defaultdict(list)
+    for secret in file_results:
+        secret_key = f'{secret.filename}_{secret.secret_hash}_{secret.type.replace(" ", "-")}'
+        equal_secret_in_commit[secret_key].append('added' if secret.is_added else 'removed')
+
+    for secret in file_results:
+        secret_key = f'{secret.filename}_{secret.secret_hash}_{secret.type.replace(" ", "-")}'
+        if not all(value in equal_secret_in_commit[secret_key] for value in ['added', 'removed']):
+            if secret.is_added:
+                if secret_key not in secret_map:
+                    secret_map[secret_key] = []
+                else:
+                    all_removed = all(
+                        potential_secret.get('removed_commit_hash') for potential_secret in secret_map[secret_key])
+                    # Update secret map with the new potential secret
+                    if all_removed:
+                        secret_map[secret_key][0].update({'potential_secret': secret, 'removed_commit_hash': ''})
+                        continue
+                secret_map[secret_key].append(
+                    {'added_commit_hash': commit_hash, 'removed_commit_hash': '', 'potential_secret': secret})
```

Comment:
Just a suggestion but maybe we can extract this block to a new function? will make the code clearer IMO